index.js

```diff
@@ -0,0 +1,187 @@
+  } else {
+    tasks = JSON.parse(localStorage.getItem('tasks'));
+  }
+  return tasks;
+};
+
+const addNewTask = (task) => {
+  const tasks = getLocalStorage();
+  tasks.push(task);
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+};
+
+const deleteTask = (index) => {
+  const tasks = getLocalStorage();
+  tasks.splice(index, 1);
+
+  for (let i = 0; i < tasks.length; i += 1) {
+    tasks[i].index = i + 1;
+  }
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+  window.location.reload();
+};
+
+const editTask = (newtext, index) => {
+  const tasks = getLocalStorage();
+  // if new text is empty delete
+  if (newtext === '') {
+    deleteTask(index);
+  } else {
+    tasks[index].description = newtext;
+    localStorage.setItem('tasks', JSON.stringify(tasks));
+  }
+};
+
+// Interactive functions
+const markCompleted = (index) => {
+  const tasks = getLocalStorage();
+  tasks[index].completed = true;
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+};
+
+const markUnCompleted = (index) => {
+  const tasks = getLocalStorage();
+  tasks[index].completed = false;
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+};
+
+const clearCompleted = () => {
+  const tasks = getLocalStorage();
+  const newtasks = tasks.filter((task) => task.completed === false);
+
+  for (let i = 0; i < newtasks.length; i += 1) {
+    newtasks[i].index = i + 1;
+  }
+  localStorage.setItem('tasks', JSON.stringify(newtasks));
+  window.location.reload();
+};
+
+// get ul from index.html
+const ul = document.getElementById('list-items');
+
+const addTaskToList = (task) => {
+  const taskItem = document.createElement('li');
+  let checkmark;
+  let completedClass;
+  if (task.completed === true) {
+    checkmark = '<span class="material-icons checkmark">done</span>';
+    completedClass = 'completed';
+  }
+  taskItem.classList.add('todo');
+  taskItem.innerHTML = `
+  <button type="button" class="checkbox ${completedClass}" >${checkmark}</button>
+  <p contentEditable="true" class="desc">${task.description}</p>
+  <div class="dots">
+    <span class="material-icons">more_vert</span>
+  </div>
+  <div class="bin">
+    <span class="material-symbols-outlined delete-bin">delete</span>
+  </div>
+  `;
+
+  ul.appendChild(taskItem);
+};
+
+const displayTasks = () => {
+  const tasks = getLocalStorage();
+  tasks.forEach((task) => addTaskToList(task));
+};
+
+// Display local storage items on page
+document.addEventListener('DOMContentLoaded', displayTasks);
+
+document.querySelector('#form').addEventListener('submit', (e) => {
+  e.preventDefault();
+
+  const description = document.querySelector('#description').value;
+
+  if (description === '') {
+    return;
+  }
+  const newTask = {};
+  newTask.description = description;
+  newTask.completed = false;
+  newTask.index = getLocalStorage().length + 1;
+  addNewTask(newTask);
+  addTaskToList(newTask);
+  document.querySelector('#description').value = '';
+  window.location.reload();
```

Comment:
Kindly fix issues with newly created elements, so you don't have to reload the page when you create a new task